interface-impl-amf/src/main/java/org/mule/amf/impl/model/ParameterImpl.java

```diff
@@ -248,4 +260,14 @@ private static boolean hasAnArrayVariant(UnionShape unionShape) {
 
     return hasAnArrayVariant;
   }
+
+  private static boolean hasNilShape(UnionShape unionShape) {
+    for (Shape shape : unionShape.anyOf()) {
+      if (shape instanceof NilShape) {
+        return true;
+      }
+    }
+    return unionShape.anyOf().stream().anyMatch(NilShape.class::isInstance);
```

Comment:
Please choose one of the options (the single line stream chain or the multiline for-loop). This code is going through the union variants twice 😛.